git-config/src/values.rs

```diff
@@ -273,6 +281,68 @@ impl Serialize for Value<'_> {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[allow(missing_docs)]
+pub struct Path<'a> {
+    pub value: Cow<'a, StdPath>,
+}
+
+impl<'a> TryFrom<&'a str> for Path<'a> {
+    type Error = String;
+
+    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
+        let path = StdPath::new(value);
+        if !path.exists() {
```

Comment:
Where does the requirement for paths needing to exist come from?

Reply:
like you said to be able to distinguish from str but yes its IO